Send notification as AdaptiveCards

Navya-Teja-Otturu opened the pull request:
According to this [link](https://devblogs.microsoft.com/microsoft365dev/retirement-of-office-365-connectors-within-microsoft-teams/), the Incoming Webhooks are deprecated and will be decommissioned. In place of Incoming webhooks workflows can be used to post notifications to MS Teams.

The branch V2(or V2 version of this action) is intended to send notifications as AdaptiveCards which uses workflows.

The Adaptive card notifications are more user appealing and looks nicer.

This Pull Request also removes the unnecessary inputs `workflow_run_success` and `workflow_run_failure` because the process of when to notify can be managed with the help of the conditions at step level or job level in the workflow.

This pull request also introduces are new input parameter `conclusion` to which the workflow conclusion
( `${{ github.workflow.conclustion}}` ) must be passed and based on the conclusion the notification is changed.

Navya-Teja-Otturu committed:
Use only AdaptiveCard type notifications

The AdaptiveCards are more user appealing and looks nicer

Navya-Teja-Otturu committed:
Prepare notification based on workflow conclusion

Navya-Teja-Otturu committed:
Document the changes in notification action

Navya-Teja-Otturu committed:
Merge branch 'main' into send_notification_using_workflows_also

Uzlopak commented on src/utils.ts:
```diff
@@ -1,142 +1,14 @@
-    text: `event: ${ctx.eventName}`,
+export const getConfig = (): Config => {
+  const result: Config = {
+    webhook_url: getInput("webhook_url"),
```
Do we validate webhook_url somewhere?

Navya-Teja-Otturu committed:
Rename the function that reads inputs

This makes code more readable

Navya-Teja-Otturu committed:
Correct the text in the heading of notification

Some events like `schedule` are not triggered by any user hence the
heading of the notification must not use `sender.login` all the time.
